src/rules/no-empty-fields.ts

```diff
@@ -0,0 +1,105 @@
+import { createRule } from "../createRule";
+import { isString, omit } from "../utils/predicates";
+import detectNewline from "detect-newline";
+import detectIndent from "detect-indent";
+
+export const rule = createRule({
+	create(context) {
+		const objectFields = [
+			"peerDependencies",
+			"scripts",
+			"dependencies",
+			"devDependencies",
+		];
+		const arrayFields = ["files"];
+
+		return {
+			"Program:exit"() {
+				const { ast, text } = context.sourceCode;
+
+				const json = JSON.parse(text) as Record<string, unknown>;
+				const { properties } = ast.body[0].expression;
+
+				for (let i = 0; i < properties.length; i += 1) {
+					const property = properties[i];
+
+					if (
+						property.key.type === "JSONLiteral" &&
+						isString(property.key.value) &&
+						objectFields.includes(property.key.value) &&
+						property.value.type === "JSONObjectExpression" &&
+						!property.value.properties.length
+					) {
+						const field = property.key.value;
+						const { indent, type } = detectIndent(text);
+						const endCharacters = text.endsWith("\n") ? "\n" : "";
+						const newline = detectNewline.graceful(text);
+						let result =
+							JSON.stringify(
+								omit(json, [field]),
+								null,
+								type === "tab" ? "\t" : indent,
+							) + endCharacters;
+						if (newline === "\r\n") {
+							result = result.replace(/\n/g, newline);
+						}
+						context.report({
+							data: {
+								property: field,
+							},
+							fix(fixer) {
+								return fixer.replaceText(ast, result);
```

Comment:
[Bug] This replaces the _entire_ `package.json` with the stringified result. Some older rules in this repo do that but it's not a great user experience:

* The whole file gets re-formatted, which is inconvenient to see in general and especially inconvenient if you don't also use a formatter
* It slows linting down if multiple rules have fixes because they can't both be applied

See how other rules perform `fixer.remove` on individual nodes. `no-redundant-files` has some nicely commented code in its reports.